README.rst

```diff
@@ -334,6 +335,26 @@ environment variable to control is ``PREMAILER_CACHE_MAXSIZE``. This parameter
 requires a little bit of fine-tuning and calibration if your workload is really
 big and memory even becomes an issue.
 
+Advanced options
+----------------
+
+Below are some advanced configuration options that probably doesn't matter for
+most people with regular load.
+
+Choosing the cache implementation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, ``premailer`` uses `LRUCache
+<https://cachetools.readthedocs.io/en/latest/#cachetools.LRUCache>`__ to cache
+selectors, styles and parsed CSS strings. If LRU doesn't serve your purpose, it
+is possible to switch to an alternate implementation using below environment
+variables.
+
+- ``PREMAILER_CACHE``: Can be LRU, LFU or TTL. Default is LRU.
+- ``PREMAILER_CACHE_MAXSIZE``: Maximum no. of items to be stored in cache. Defaults to 128.
```

Comment:
I wonder, that 128 comes from https://docs.python.org/3/library/functools.html#functools.lru_cache I think. Who has a HTML document, with CSS that has 128 or less selectors?! Wouldn't it make sense to bump this up to at least 1,000?

Another option is to set it to `None` but I think people would be very negatively surprised if their memory explodes. 


By the way, they say, for `functools.lru_cache`:

> The LRU feature performs best when maxsize is a power-of-two.

I'm not sure how that applies to `cachetools`, can you find out, but then the better default would be 1024.

Reply:
I looked into [LRU implementation](https://github.com/python/cpython/blob/master/Lib/functools.py#L469) and couldn't figure out why they suggest the `maxsize to be power of two`.

One drawback of keeping higher maxsize that I see is that cache is cleared less frequently even if items in cache are never used (unless you are using TTL cache ofcourse) and with `cache_css_parsing` turned on, the memory usage can go really high.

I would like to leave it to default of 128 and let users choose appropriate value if 128 is too less for them.

Let me know your thoughts.

Reply:
Fine by me. 

I suspect that many users of `premailer` use it sporadically. E.g. when a form is submitted, an email is sent. 

If you're using it for big bulk processing, like you, you come here and learn about the max_size option.